### ISSUE TITLE
deleting an email

### ISSUE THREAD

**Issue by olivierharo** [jamesiarmes/php-ews#378]
Version (e.g. 1.0, dev-master):
Microsoft Exchange version: 2013

Hello, i'm trying to delete an Item (inbox email) , but getting an error message.

REQUEST : 

$client = new Client($host, $username, $password, $version);
$client->setTimezone('Eastern Standard Time');
$request = new DeleteItemType();
$request->DeleteType='HardDelete';
$request->ItemIds->ItemId[Id]="AAMkAGNjMTFhNDZkLThmMWYtNGJhZS04OGNkLWU4Nzg1NjQzOTZiNQBGAAAAAACVRr76ckE4QoED6dzqH3d6BwCy9f2x/KW6SIigd5LEhwq7AA0jCQE6AACy9f2x/KW6SIigd5LEhwq7ACEWCZt9AAA=";
$request->ItemIds->ItemId[ChangeKey]="CQAAABYAAACy9f2x/KW6SIigd5LEhwq7ACEWDBGM";
// exp : saroumane67@gmail.com : 
print_r($request);
$response = $client->DeleteItem($request);

RESPONSE : 

The request failed schema validation: The element 'TimeZoneContext' in namespace 'http://schemas.microsoft.com/exchange/services/2006/types' has invalid child element 'item'.

**Comment by jamesiarmes** [comment]
Try without the `$client->setTimezone('Eastern Standard Time');` line. Delete operations may not support the TimeZoneContext header.

**Comment by talalong** [comment]
can I do the same thing in order to delete my events.
this below code does not work.
```
$client = new Client($host, $username, $password, $version);
$client->setCurlOptions(array(CURLOPT_SSL_VERIFYPEER => false));

$request = new DeleteItemType();
$request->DeleteType = DisposalType::HARD_DELETE;

$request->ItemIds->ItemId="AAMkAGNjMTFhNDZkLThmMWYtNGJhZS04OGNkLWU4Nzg1NjQzOTZiNQBGAAAAAACVRr76ckE4QoED6dzqH3d6BwCy9f2x/KW6SIigd5LEhwq7AA0jCQE6AACy9f2x/KW6SIigd5LEhwq7ACEWCZt9AAA=";
$response = $client->DeleteItem($request);
```

**Comment by jamesiarmes** [comment]
@talalong How does the code not work? Does it throw an exception? Does Exchange return an error?

**Comment by talalong** [comment]
@jamesiarmes Exchange returns me an `DeleteItemResponse` object with a message 
> The `SendMeetingCancellations ` attribute is required for calendar items
And the server returns an error on this line `$request->ItemIds->ItemId = new ItemIdType();`
> Creating default object from empty

Full response:

> object(jamesiarmes\PhpEws\Response\DeleteItemResponseType)#8 (1) { ["ResponseMessages"]=> object(jamesiarmes\PhpEws\ArrayType\ArrayOfResponseMessagesType)#9 (54) { ["ApplyConversationActionResponseMessage"]=> array(0) { } ["ArchiveItemResponseMessage"]=> array(0) { } ["ConvertIdResponseMessage"]=> array(0) { } ["CopyFolderResponseMessage"]=> array(0) { } ["CopyItemResponseMessage"]=> array(0) { } ["CreateAttachmentResponseMessage"]=> array(0) { } ["CreateFolderResponseMessage"]=> array(0) { } ["CreateItemResponseMessage"]=> array(0) { } ["CreateManagedFolderResponseMessage"]=> array(0) { } ["CreateUserConfigurationResponseMessage"]=> array(0) { } ["DeleteAttachmentResponseMessage"]=> array(0) { } ["DeleteFolderResponseMessage"]=> array(0) { } ["DeleteItemResponseMessage"]=> array(1) { [0]=> object(jamesiarmes\PhpEws\Response\ResponseMessageType)#10 (5) { ["DescriptiveLinkKey"]=> int(0) ["MessageText"]=> string(79) "Das Attribut "SendMeetingCancellations" ist für Kalenderelemente erforderlich." ["MessageXml"]=> NULL ["ResponseClass"]=> string(5) "Error" ["ResponseCode"]=> string(37) "ErrorSendMeetingCancellationsRequired" } } ["DeleteUserConfigurationResponseMessage"]=> array(0) { } ["EmptyFolderResponseMessage"]=> array(0) { } ["ExpandDLResponseMessage"]=> array(0) { } ["ExportItemsResponseMessage"]=> array(0) { } ["FindFolderResponseMessage"]=> array(0) { } ["FindItemResponseMessage"]=> array(0) { } ["FindMailboxStatisticsByKeywordsResponseMessage"]=> array(0) { } ["GetAttachmentResponseMessage"]=> array(0) { } ["GetClientAccessTokenResponseMessage"]=> array(0) { } ["GetConversationItemsResponseMessage"]=> array(0) { } ["GetEventsResponseMessage"]=> array(0) { } ["GetFolderResponseMessage"]=> array(0) { } ["GetItemResponseMessage"]=> array(0) { } ["GetRemindersResponse"]=> array(0) { } ["GetRoomListsResponse"]=> array(0) { } ["GetRoomsResponse"]=> array(0) { } ["GetServerTimeZonesResponseMessage"]=> array(0) { } ["GetSharingFolderResponseMessage"]=> array(0) { } ["GetSharingMetadataResponseMessage"]=> array(0) { } ["GetStreamingEventsResponseMessage"]=> array(0) { } ["GetUserConfigurationResponseMessage"]=> array(0) { } ["GetUserPhotoResponseMessage"]=> array(0) { } ["MarkAllItemsAsReadResponseMessage"]=> array(0) { } ["MarkAsJunkResponseMessage"]=> array(0) { } ["MoveFolderResponseMessage"]=> array(0) { } ["MoveItemResponseMessage"]=> array(0) { } ["PerformReminderActionResponse"]=> array(0) { } ["RefreshSharingFolderResponseMessage"]=> array(0) { } ["ResolveNamesResponseMessage"]=> array(0) { } ["SearchMailboxesResponseMessage"]=> array(0) { } ["SendItemResponseMessage"]=> array(0) { } ["SendNotificationResponseMessage"]=> array(0) { } ["SetHoldOnMailboxesResponseMessage"]=> array(0) { } ["SubscribeResponseMessage"]=> array(0) { } ["SyncFolderHierarchyResponseMessage"]=> array(0) { } ["SyncFolderItemsResponseMessage"]=> array(0) { } ["UnsubscribeResponseMessage"]=> array(0) { } ["UpdateFolderResponseMessage"]=> array(0) { } ["UpdateItemResponseMessage"]=> array(0) { } ["UpdateUserConfigurationResponseMessage"]=> array(0) { } ["UploadItemsResponseMessage"]=> array(0) { } } }

**Comment by talalong** [comment]
Do I actually need `$request->SendMeetingCancellations = EWSType_CalendarItemCreateOrDeleteOperationType::SEND_TO_NONE;`

**Comment by jamesiarmes** [comment]
Yes, you need to set `SendMeetingCancellations` when deleting an event. The warning you're getting regarding the default object is because you aren't setting `$request->ItemIds` before using it.

**Comment by talalong** [comment]
It works. Thank you ;)

**Comment by jamesiarmes** [comment]
@olivierharo Can you tell me if your issue has been resolved?
